### PR TITLE
Update some create `has_one` failure test descriptions

### PR DESCRIPTION
**Pull request by joshuay03**
### Motivation / Background

Update some create `has_one` test descriptions to indicate failure when using the `#create_*` constructor.

### Detail

I ran into these tests when looking into a separate issue and found the failures confusing when running against my changes. The descriptions were change [in this PR](https://github.com/rails/rails/pull/41238), but I think the reasoning there is incorrect. These are indeed creation failures, and I have added an additional assertion to make that clear.

### Additional information

None.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [ ] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: activerecord/test/cases/associations/has_one_associations_test.rb [hunk 1]
```diff
@@ -531,7 +531,7 @@ def test_attributes_are_being_set_when_initialized_from_has_one_association_with
     assert_equal "Account", new_account.firm_name
   end
 
-  def test_create_association_replaces_existing_without_dependent_option
+  def test_create_association_failure_replaces_existing_without_dependent_option
```
Comment (p8) [inline activerecord/test/cases/associations/has_one_associations_test.rb:534]: Should it be "test_creation_failure_replaces...." instead? Similar to line 558 and how it was before?
```suggestion
  def test_creation_failure_replaces_existing_without_dependent_option
```

File: activerecord/test/cases/associations/has_one_associations_test.rb [hunk 2]
```diff
@@ -540,16 +540,18 @@ def test_create_association_replaces_existing_without_dependent_option
     assert_not_equal ships(:black_pearl), new_ship
     assert_equal new_ship, pirate.ship
     assert_predicate new_ship, :new_record?
+    assert_predicate new_ship, :invalid?
     assert_nil orig_ship.pirate_id
     assert_not orig_ship.changed? # check it was saved
   end
 
-  def test_create_association_replaces_existing_with_dependent_option
+  def test_create_association_failure_replaces_existing_with_dependent_option
```
Comment (p8) [inline activerecord/test/cases/associations/has_one_associations_test.rb:548]:
```suggestion
  def test_creation_failure_replaces_existing_with_dependent_option
```

File: activerecord/test/cases/associations/has_one_associations_test.rb (hunk 1 above)
Comment (joshuay03) [reply activerecord/test/cases/associations/has_one_associations_test.rb]: Ya I guess association is already implied, I've updated 👍🏽

**Comment by joshuay03** [comment]
Labels are incorrect cause I fumbled my rebase 😅

File: activerecord/test/cases/associations/has_one_associations_test.rb [hunk 3]
```diff
@@ -540,16 +540,18 @@ def test_create_association_replaces_existing_without_dependent_option
     assert_not_equal ships(:black_pearl), new_ship
     assert_equal new_ship, pirate.ship
     assert_predicate new_ship, :new_record?
+    assert_predicate new_ship, :invalid?
```
Comment (p8) [inline activerecord/test/cases/associations/has_one_associations_test.rb:543]: Maybe assert for the error here for clarity? Similar to line 655.

Comment (joshuay03) [reply activerecord/test/cases/associations/has_one_associations_test.rb:543]: I've addressed.

Comment (p8) [reply activerecord/test/cases/associations/has_one_associations_test.rb:543]: Hmm, looking at it again with fresh eyes I think just the following is good enough:
```
assert_predicate new_ship, :invalid?
```
Sorry 😞 
Could you remove the assertion on the errors again?
After that I think this is good to go.

Comment (joshuay03) [reply activerecord/test/cases/associations/has_one_associations_test.rb:543]: Removed [here](https://github.com/rails/rails/compare/045907cc63f0f4bdf59367349c6adb86921fef8f..e61234dac78f5a77712b853b2ccf5c3963a54a06).

**Review by p8 — APPROVED** [review]

**Comment by p8** [comment]
Thanks @joshuay03

**Pull request merged**